File: Source/DotNET/Model/Common/Thickness.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Scene.Model.Common;

/// <summary>
/// Represents a uniform or per-edge thickness, used for margins and padding.
/// </summary>
/// <param name="Left">The left edge.</param>
/// <param name="Top">The top edge.</param>
/// <param name="Right">The right edge.</param>
/// <param name="Bottom">The bottom edge.</param>
public record Thickness(double Left = 0, double Top = 0, double Right = 0, double Bottom = 0);

File: Source/JavaScript/model/common/Thickness.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

/**
 * Represents a uniform or per-edge thickness, used for margins and padding.
 */
export interface Thickness {
    left: number;
    top: number;
    right: number;
    bottom: number;
}

export const ThicknessPropertyNames: (keyof Thickness)[] = ['left', 'top', 'right', 'bottom'];
